### ISSUE TITLE
diseqc issue ?

### ISSUE THREAD

**Issue by 9000h** [catalinii/minisatip#491]
find this one after some switching between dvb-s, dvb-c,dvb-t2
```
[18/11 18:23:59.898 main]: copy_dvb_param start -> src=0, fe=0, freq=466000, fec=9, sr=6900000, pol=0, ro=3, msys=1, mtype=5, plts=2, bw=8000000, inv=2, pids=NULL, apids=101, dpids=116 x_pmt=NULL
[18/11 18:23:59.898 main]: copy_dvb_parameters -> src=1, fe=0, freq=11302000, fec=2 sr=22000000, pol=2, ro=0, msys=6, mtype=9, plts=2, bw=8000000, inv=2, pids=NULL, apids=NULL, dpids=NULL x_pmt=NULL
[18/11 18:23:59.898 main]: send_diseqc fd 19, pos = 0 (c 0 u 0), pol = 1, hiband = 0
[18/11 18:23:59.898 main]: send_diseqc: FE_SET_TONE failed for fd 19: Operation not supported
[18/11 18:23:59.898 main]: send_diseqc: FE_SET_VOLTAGE failed for fd 19: Operation not supported
[18/11 18:23:59.913 main]: send_diseqc: FE_DISEQC_SEND_MASTER_CMD committed failed for fd 19: Operation not supported
[18/11 18:23:59.998 main]: send_diseqc: FE_DISEQC_SEND_BURST failed for fd 19: Operation not supported
[18/11 18:24:00.013 main]: send_diseqc: FE_SET_TONE failed for fd 19: Operation not supported
[18/11 18:24:00.013 main]: tuning to 11302000(1552000) pol: h (2) sr:22000000 fec:23 delsys:dvbs2 mod:8psk rolloff:0.35 pilot: , ts clear=43847, ts pol=43847
[18/11 18:24:00.013 main]: dvb_tune: set property failed 22 Invalid argument
[18/11 18:24:00.013 main]: Adding pid 0 to the list of pids as not explicitly added for adapter 6
[18/11 18:24:00.013 main]: deleting PMT 0, master PMT 0, name 13th Street (VF)
```
http://paste.ubuntu.com/p/r92XFHHy5R/

**Comment by 9000h** [comment]
the logging in get_free_adapter seem to also not right as  is 0 all the time, look for sys: and the frequency  which is dvbc 

```
[18/11 18:23:25.741 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:0 pol=0 sys: dvbs dvbs2
[18/11 18:23:31.069 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:0 pol=0 sys: dvbs dvbs2
[18/11 18:23:36.427 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:11493000 pol=2 sys: dvbs dvb
[18/11 18:23:40.466 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:11361000 pol=2 sys: dvbs dvb
[18/11 18:23:44.553 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:10921000 pol=2 sys: dvbs dvb
[18/11 18:23:49.291 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:10921000 pol=2 sys: dvbs dvb
[18/11 18:24:01.162 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:10921000 pol=2 sys: dvbs dvb
[18/11 18:24:05.466 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:11302000 pol=2 sys: dvbs dvb
[18/11 18:24:08.240 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:386000 pol=0 sys: dvbs dvbs2
[18/11 18:24:12.893 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:386000 pol=0 sys: dvbs dvbs2
[18/11 18:24:16.749 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:10971000 pol=2 sys: dvbs dvb
[18/11 18:24:32.561 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:10971000 pol=2 sys: dvbs dvb
[18/11 18:24:37.971 main]: get free adapter 0 - a[0] => e:1 m:-1 sid_cnt:0 f:466000 pol=0 sys: dvbs dvbs2
```

**Comment by 9000h** [comment]
get_adapter1 is an unused funktion

**Comment by 9000h** [comment]
could not find why it did try to tune with an msys  on an adapter with the wrong msys.

**Comment by catalinii** [comment]
Can you reproduce?

I guess the issue is compare_slave_parameters does not return 1 to trigger closing the adapter for that stream. The code is here:https://github.com/catalinii/minisatip/blob/master/src/stream.c#L272

**Comment by 9000h** [comment]
yes I can reproduce the issue at any time.
if I change the line to "if (ad || compare_slave_parameters(ad, &sid->tp))" the the issue seems to be fixed at least with 1 VDR (no 404 or tune errors)
if I have my Panasonic in parallel zapping around on the same channels VDR is using the at some point the Panasonic did freeze the picture and has no signal anymore
so seems to be half of the solution

**Comment by 9000h** [comment]
here are the log with the suggested line changed to ||
http://paste.ubuntu.com/p/PDDQ8skB2K/

**Comment by catalinii** [comment]
Can you try ad && !compare_slave.... ?

**Comment by catalinii** [comment]
I have just pushed a commit, let me know if it makes the things better

**Comment by 9000h** [comment]
did try but unfortunately there are still issues with shared access, a single client seems to be working better now
http://paste.ubuntu.com/p/hhhxqcq5mW/

**Comment by catalinii** [comment]
Can you point out the section where this happens?

**Comment by 9000h** [comment]
around "[21/11 19:02:58.413 main]: RTSP/1.0 404 Not Found"

**Comment by 9000h** [comment]
did some quick tests and the last commit seems to fix it
http://paste.ubuntu.com/p/CFSwPmZqFw/

many thanks!

**Comment by 9000h** [comment]
I do have another issue with loosing the signal on the Panasonic under some conditions which has probably a different root cause and is specific to this client only.
I will make some tcpdumps and logs and will open a new issue, but this is low prio.

**Comment by catalinii** [comment]
Let's start with the logs first...

**Comment by 9000h** [comment]
Ok, but I can do it earliest Sunday when I back.
Which log level do you prefer?

It could be also a new "feature" coming with a software update on the Panasonic :-/

**Comment by catalinii** [comment]
Same as before
